android/sdl_android/src/main/java/com/smartdevicelink/transport/TransportManager.java

```diff
@@ -78,32 +82,45 @@
     public TransportManager(MultiplexTransportConfig config, TransportEventListener listener){
         super(config,listener);
 
+        this.transportListener = listener;
+        this.TRANSPORT_STATUS_LOCK = new Object();
+        this.mConfig = config;
+        synchronized (TRANSPORT_STATUS_LOCK){
+            this.transportStatus = new ArrayList<>();
+        }
+
         if(config.service == null) {
             config.service = SdlBroadcastReceiver.consumeQueuedRouterService();
         }
 
         contextWeakReference = new WeakReference<>(config.context);
-
-        RouterServiceValidator validator = new RouterServiceValidator(config);
-        if(validator.validate()){
-            transport = new TransportBrokerImpl(config.context, config.appId,config.service);
-        }else{
-            enterLegacyMode("Router service is not trusted. Entering legacy mode");
-        }
     }
 
+    /**
+     * start internally validates the target ROuterService, which was done in ctor before.
+     */
     @Override
-    public void start(){
-        if(transport != null){
-            if (!transport.start()){
-                //Unable to connect to a router service
-                if(transportListener != null){
-                    transportListener.onError("Unable to connect with the router service");
+    public void start() {
+        final RouterServiceValidator validator = new RouterServiceValidator(mConfig);
+        validator.validateAsync(new RouterServiceValidator.ValidationStatusCallback() {
+            @Override
+            public void onFinishedValidation(boolean valid, ComponentName name) {
+                Log.d(TAG, "onFinishedValidation valid=" + valid + "; name=" + ((name == null)? "null" : name.getPackageName()));
+                if (valid) {
+                    mConfig.service = name;
+                    transport = new TransportBrokerImpl(contextWeakReference.get(), mConfig.appId, mConfig.service);
+                    Log.d(TAG, "TransportManager start got called; transport=" + transport);
+                    if(transport != null){
+                        transport.start();
```

Comment:
Well, because RouterServiceValidator checks to see if we have the connected RouterService exists, and because TransportBrokerImpl.start() here gets called only if valid == true, start() call should always succeeds.